evalml/tests/component_tests/test_knn_imputer.py

```diff
@@ -0,0 +1,162 @@
+
+
+@pytest.mark.parametrize("n_neighbors", [1, 2, 5])
+def test_knn_output(n_neighbors):
+    X = pd.DataFrame(
+        np.array(
+            [
+                [1, 1, 2],
+                [1, 1, np.nan],
+                [1, 1, 2],
+                [1, 1, 2],
+                [1, 2, 1],
+                [1, 1, 2],
+                [1, 1, 2],
+            ],
+        ),
+    )
+    sk_knn = Sk_KNNImputer(n_neighbors=n_neighbors)
+    knn = KNNImputer(n_neighbors)
+    assert_frame_equal(pd.DataFrame(sk_knn.fit_transform(X)), knn.fit_transform(X))
+
+
+@pytest.mark.parametrize("data_type", ["pd", "ww"])
+def test_knn_imputer_all_bool_return_original(data_type, make_data_type):
+    X = pd.DataFrame([True, True, False, True, True], dtype=bool)
+    y = pd.Series([1, 0, 0, 1, 0])
+    X = make_data_type(data_type, X)
+    y = make_data_type(data_type, y)
+    X_expected_arr = pd.DataFrame([True, True, False, True, True], dtype=bool)
+    imputer = KNNImputer()
+    imputer.fit(X, y)
+    X_t = imputer.transform(X)
+    assert_frame_equal(X_expected_arr, X_t)
+
+
+@pytest.mark.parametrize("data_type", ["pd", "ww"])
+def test_knn_imputer_boolean_dtype(data_type, make_data_type):
+    X = pd.DataFrame(
+        {
+            "some_nan": pd.Series([True, np.nan, False, np.nan, True], dtype="boolean"),
+        },
+    )
+    X.ww.init(logical_types={"some_nan": "BooleanNullable"})
+    y = pd.Series([1, 0, 0, 1, 0])
+    X_expected_arr = pd.DataFrame(
+        {
+            "some_nan": pd.Series([True, True, False, True, True]),
+        },
+    )
+    X = make_data_type(data_type, X)
+    imputer = KNNImputer(number_neighbors=1)
+    X_t = imputer.fit_transform(X, y)
+    X_ww_expected = "<Series: some_nan (Physical Type = bool) (Logical Type = Boolean) (Semantic Tags = set())>"
+    assert str(X_t.ww["some_nan"].ww) == X_ww_expected
+    assert_frame_equal(X_expected_arr, X_t)
+
+
+@pytest.mark.parametrize("data_type", ["pd", "ww"])
+def test_knn_imputer_multitype_with_one_bool(data_type, make_data_type):
+    X_multi = pd.DataFrame(
+        {
+            "bool with nan": pd.Series([True, np.nan, False, np.nan, False]),
+            "bool no nan": pd.Series([False, False, False, False, True], dtype=bool),
+        },
+    )
+    X_multi.ww.init(logical_types={"bool with nan": "BooleanNullable"})
+    y = pd.Series([1, 0, 0, 1, 0])
+    X_multi_expected_arr = pd.DataFrame(
+        {
+            "bool with nan": pd.Series(
+                [True, True, False, True, False],
+                dtype=bool,
+            ),
+            "bool no nan": pd.Series([False, False, False, False, True], dtype=bool),
+        },
+    )
+    X_multi = make_data_type(data_type, X_multi)
+
+    imputer = KNNImputer(number_neighbors=1)
+    imputer.fit(X_multi, y)
+    X_multi_t = imputer.transform(X_multi)
+    assert_frame_equal(X_multi_expected_arr, X_multi_t)
+
+
+def test_knn_imputer_revert_categorical_to_boolean():
```

Comment:
is this test to check if `BooleanNullables` will be converted to `Boolean`? If so let's parameterize this method to use pandas and ww as well and then check the logical types in the WW case.

Reply:
Line 92 was to test if I could revert categorical to boolean. However KNNImputer doesn't have the same issues that SimpleImputer has if you pass in just bools (no nulls), therefore I can get rid of line 92's test (since I don't have to actually change bool columns that are all bools into categorical).